scrabble.js

```diff
@@ -1,8 +1,204 @@
+                            score: currentScore
+          }
+        };
+      }else if (currentScore > highestScoreHash["score"]){
+        highestScoreHash = {
+                          word: word,
+                          score: currentScore
+        }
+      };
+    };
+    return highestScoreHash["word"];
+  };
+
+// PLAYER
+  Player = function(name){
+    this.name = name;
+    this.plays = [];
+    this.totalScore = 0;
+    this._tiles = [];
+    this._maxTiles = 7;
+    this._scrabble = new Scrabble();
+  };
+
+  Player.prototype.play = function(word){
+    var score = this._scrabble.score(word);
+
+    this.plays.push(word);
+    this.totalScore += score;
+
+    return this.hasWon() === true ? false : score;
+  };
+
+  Player.prototype.hasWon = function(){
+    return this.totalScore >= 100 ? true : false;
+  };
+
+  Player.prototype.highestScore = function(){
+    return this.maxByScore()["score"];
+  };
+
+  Player.prototype.highestScoringWord = function(){
+    return this.maxByScore()["word"];
+  };
+
+
+  Player.prototype.maxByScore = function(){
+    var initialWord = this.plays[0];
+    var maxHash = {word: initialWord, score: this._scrabble.score(initialWord)};
+
+    for(var word of this.plays){
+      var currentScore = this._scrabble.score(word);
+      if(currentScore > maxHash["score"]){
+        maxHash = {word: word, score: currentScore};
+      };
+    };
+    return maxHash;
+  };
+
+
+  TileBag = function(){
+    this._defaultTiles = [["A", 1], ["A", 1], ["A", 1], ["A", 1], ["A", 1], ["A", 1], ["A", 1], ["A", 1], ["A", 1],
+                  ["B", 1], ["B", 1],
+                  ["C", 1], ["C", 1],
+                  ["D", 1], ["D", 1], ["D", 1], ["D", 1],
+                  ["E", 1], ["E", 1], ["E", 1], ["E", 1], ["E", 1], ["E", 1], ["E", 1],
+                  ["E", 1], ["E", 1], ["E", 1], ["E", 1], ["E", 1],
+                  ["F", 1], ["F", 1],
+                  ["G", 1], ["G", 1],["G", 1],
+                  ["H", 1], ["H", 1],
+                  ["I", 1], ["I", 1], ["I", 1], ["I", 1], ["I", 1], ["I", 1], ["I", 1], ["I", 1], ["I", 1],
+                  ["J", 1],
+                  ["K", 1],
+                  ["L", 1], ["L", 1], ["L", 1], ["L", 1],
+                  ["M", 1], ["M", 1],
+                  ["N", 1], ["N", 1], ["N", 1], ["N", 1], ["N", 1], ["N", 1],
+                  ["O", 1], ["O", 1], ["O", 1], ["O", 1], ["O", 1], ["O", 1], ["O", 1], ["O", 1],
+                  ["P", 1], ["P", 1],
+                  ["Q", 1],
+                  ["R", 1], ["R", 1], ["R", 1], ["R", 1], ["R", 1], ["R", 1],
+                  ["S", 1], ["S", 1], ["S", 1], ["S", 1],
+                  ["T", 1], ["T", 1], ["T", 1], ["T", 1], ["T", 1], ["T", 1],
+                  ["U", 1], ["U", 1], ["U", 1], ["U", 1],
+                  ["V", 1], ["V", 1],
+                  ["W", 1], ["W", 1],
+                  ["X", 1],
+                  ["Y", 1], ["Y", 1],
+                  ["Z", 1]];
+  };
+
```

Comment:
lol I cheated. I wrote code that spitted it out. hehehe

On Wed, Jun 15, 2016 at 9:48 PM, Jennie Buechner <notifications@github.com>
wrote:

> In scrabble.js
> <https://github.com/Ada-C5/js-scrabble/pull/17#discussion_r67286942>:
>
> > +                  ["M", 1], ["M", 1],
> > +                  ["N", 1], ["N", 1], ["N", 1], ["N", 1], ["N", 1], ["N", 1],
> > +                  ["O", 1], ["O", 1], ["O", 1], ["O", 1], ["O", 1], ["O", 1], ["O", 1], ["O", 1],
> > +                  ["P", 1], ["P", 1],
> > +                  ["Q", 1],
> > +                  ["R", 1], ["R", 1], ["R", 1], ["R", 1], ["R", 1], ["R", 1],
> > +                  ["S", 1], ["S", 1], ["S", 1], ["S", 1],
> > +                  ["T", 1], ["T", 1], ["T", 1], ["T", 1], ["T", 1], ["T", 1],
> > +                  ["U", 1], ["U", 1], ["U", 1], ["U", 1],
> > +                  ["V", 1], ["V", 1],
> > +                  ["W", 1], ["W", 1],
> > +                  ["X", 1],
> > +                  ["Y", 1], ["Y", 1],
> > +                  ["Z", 1]];
> > +  };
> > +
>
> I bet it was fun typing all that out 😛
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ada-C5/js-scrabble/pull/17/files/68f6a6ae8a0ccd97a9fe4f3a205163ad6eefb2ee#r67286942>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AOaZbc7BelRsgGX-7K6HI1rmAJP0iXlnks5qMNWwgaJpZM4Iz-uw>
> .
>